content/300-guides/100-prisma-guides/240-seed-database.mdx

```diff
@@ -0,0 +1,169 @@
+---
+title: Seeding your database with Prisma Client
+metaDescription: Learn how to seed your database using Prisma Client
+---
+
+<TopBlock>
+
+This guide describes how to seed your database using Prisma Client. This makes it really easy to ensure you can replay the necessary data in your database as often as needed. You may want to do this for multiple reasons:
+
+* To populate databases with the initial set of data necessary for your application to run
+* During development, so you have some basic data for validating and using your application. This is especially helpful when using Prisma Migrate, since your development database may sometimes be reset. 
+
+</TopBlock>
+
+## Seeding in TypeScript
+
+If you are using TypeScript, follow these steps:
+
+1. Create a new file `prisma/seeds.ts`
+2. In the `prisma/seeds.ts` file, import Prisma Client, initialize it and create some records. Example:
+
+    ```ts
+    import { PrismaClient } from '@prisma/client'
+
+    const prisma = new PrismaClient()
+
+    async function main() {
+    const alice = await prisma.user.upsert({
+        where: { email: 'alice@prisma.io' },
+        update: {},
+        create: {
+        email: `alice@prisma.io`,
+        name: 'Alice',
+        posts: {
+            create: {
+            title: 'Check out Prisma with Next.js',
+            content: 'https://www.prisma.io/nextjs',
+            published: true,
+            },
+        },
+        },
```

Comment:
Seems that the code indentation is weird, see source https://github.com/Jolg42/prisma-seed-example/blob/main/typescript/prisma/seed.ts

Reply:
Applies to both code examples in multiples spots